# HG changeset patch
# User Keir Fraser <keir.fraser@citrix.com>
# Date 1225190351 0
# Node ID de7f94bd650b7e00cd57191280c5f0959b2a286e
# Parent  f40f4f86d5a2c5caa0261512279f3590e95f3d91
dom0: Fix for throttling while pr->id == -1

Signed-off-by: Wei Gang <gang.wei@intel.com>

diff -r f40f4f86d5a2 -r de7f94bd650b drivers/acpi/processor_core.c
--- a/drivers/acpi/processor_core.c	Mon Oct 27 13:47:07 2008 +0000
+++ b/drivers/acpi/processor_core.c	Tue Oct 28 10:39:11 2008 +0000
@@ -677,8 +677,17 @@
 #if defined(CONFIG_CPU_FREQ) || defined(CONFIG_PROCESSOR_EXTERNAL_CONTROL)
 	acpi_processor_ppc_has_changed(pr);
 #endif
-	acpi_processor_get_throttling_info(pr);
-	acpi_processor_get_limit_info(pr);
+
+	/* 
+	 * pr->id may equal to -1 while processor_cntl_external enabled.
+	 * throttle and thermal module don't support this case.
+	 * Tx only works when dom0 vcpu == pcpu num by far, as we give 
+	 * control to dom0.
+	 */
+	if (pr->id != -1) {
+		acpi_processor_get_throttling_info(pr);
+		acpi_processor_get_limit_info(pr);
+	}
 
 
 	acpi_processor_power_init(pr, device);
